src/Filter/ORM/AbstractFilter.php

```diff
@@ -34,10 +35,15 @@ public function getFilterManager()
 
     protected function typeCastField($metadata, $field, $value, $format, $doNotTypecastDatetime = false)
     {
-        if (! isset($metadata->fieldMappings[$field])) {
+        // If association just return
+        if (isset($metadata->associationMappings[$field])) {
```

Comment:
Hm... and here we are... It is not gonna work if we have more complex relation. For example:
Artist->Album->Song
and I have to fetch all songs of some artist, `GET /song` with:
```
'filters' => [
    [
        'type' => 'innerjoin',
        'field' => 'album',
        'alias' => 'album',
    ],
    [
        'type' => 'innerjoin',
        'parentAlias' => 'album',
        'field' => 'artist',
        'alias' => 'artist',
    ],
    [
        'type' => 'eq',
        'field' => 'name',
        'value' => 'webimpress',
    ],
];
```

And there is no field "name" in `song` table ;-)

There is also one example in docs with joining tables from joined tables:
https://github.com/zfcampus/zf-doctrine-querybuilder#joining-entities-and-aliasing-queries